thoth/sourcemanagement/sourcemanagement.py

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env python3
+# Kebechet
+# Copyright(C) 2018, 2019, 2020 Sai Sankar Gochhayat, Fridolin Pokorny
+#
+# This program is free software: you can redistribute it and / or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+"""Abstract calls to GitHub and GitLab APIs."""
+
+import logging
+import typing
+
+import requests
+from urllib.parse import quote_plus
+
+from ogr.services.github import GithubService
+from ogr.services.gitlab import GitlabService
+
+from .enums import ServiceType
+from ogr.abstract import Issue
+from ogr.abstract import PullRequest
+from ogr.abstract import PRStatus
+from .exception import CannotFetchPRError
+from .exception import CannotFetchBranchesError
+from .exception import CreatePRError
+import json
+
+_LOGGER = logging.getLogger(__name__)
+BASE_URL = {"github": "https://api.github.com", "gitlab": "https://gitlab.com//api/v4"}
+
+
+class SourceManagement:
+    """Abstract source code management services like GitHub and GitLab."""
+
+    def __init__(
+        self, service_type: ServiceType, service_url: str, token: str, slug: str
+    ):
+        """Initialize source code management tools abstraction.
+
+        Note that we are using OGR for calls. OGR keeps URL to services in its global context per GitHub/GitLab.
+        This is global context is initialized in the manager with a hope to fix this behavior for our needs.
+        """
+        self.service_type = service_type
```

Comment:
It might make sense to pass service type as a string — resolve it to an enum during `init`.  That way users don't have to include `sourcemanagement.enums` every time. @fridex LMT what you think

Reply:
From kebechet point of view, we need to convert from ServiceType to string, and then pass it to sourcemanagement. So that's why I put the ServiceType in sourcemanagement.

Reply:
To have safe API, its better to use enums.